pkg/network/setup/netconf.go

```diff
@@ -29,9 +29,14 @@ import (
 	"kubevirt.io/kubevirt/pkg/network/cache"
 )
 
+type cacheCreator interface {
```

Comment:
Why not making [this](https://github.com/kubevirt/kubevirt/pull/7023/files#diff-899c8e2d0c4b77be920098631c642717964f28ec64d28d3210ad76b20ea04a59R86) one public?

Reply:
It is not a good practice to define the interface that **others** use in the package the implements it. It creates an unwanted dependency between the interface and the concrete (or has a potential to do so).

The recommended best practice is to define the interface at the usage or in a separated package (e.g. [io](https://pkg.go.dev/io)).
I do not think it deserved its own package, nor I have other things to put there.

The end result is that at the usage, there is no direct dependency on the concrete.
I do see the annoying point where we need to define it in multiple places, but in the balance of using interfaces correctly, I though it is worth the duplication and the explicit definition of them as "private".
(just to make this clear, if we change the interface here to add more behavior, then we do not need to touch the other usages at all).